pkgs/by-name/in/installShellFiles/setup-hook.sh

```diff
@@ -233,3 +233,26 @@ installShellCompletion() {
     fi
     return $retval
 }
+
+# installBin <path> [...<path>]
+#
+# Install each argument to $outputBin
+installBin() {
+    local path
+    for path in "$@"; do
+        if (( "${NIX_DEBUG:-0}" >= 1 )); then
+            echo "installBin: installing $path"
+        fi
+        if test -z "$path"; then
+            echo "installBin: error: path cannot be empty" >&2
+            return 1
+        fi
+        local basename
+        basename=$(stripHash "$path") # use stripHash in case it's a nix store path
+
+        local outRoot
+        outRoot=${!outputBin:?}
+
+        install -Dm755 -T "$path" "${outRoot}/bin/$basename" || return
```

Comment:
I don't see this "succeed on error" on other things. Why this one?

Reply:
Well, this is the same construction on `installManPage`:

https://github.com/NixOS/nixpkgs/pull/332612/files/ce45e04e92ce8f22fb281db1d9d2c9a84bf98de5#diff-1bf14a6bd9bd0976c8094adb934eefa32d7904ce6ef119a7dae5cd6e8fbfc63eL48

Reply:
It was introduced in https://github.com/NixOS/nixpkgs/commit/43dade238f39f. @matthewbauer had some commentary on that commit.

Note that `installManPage` is the only one with that questionable pattern. I'm pretty against this error swallowing behavior. There's no rationale in the original commit, and it's liable to look right and be wrong, which is ... not great.

Reply:
I rewrote the functions to be more conformant.